components/public-api-server/pkg/oidc/service.go

```diff
@@ -152,7 +150,7 @@ func (s *Service) GetClientConfigFromCallbackRequest(r *http.Request) (*ClientCo
 		return nil, fmt.Errorf("missing state parameter")
 	}
 
-	state, err := decodeStateParam(stateParam)
+	state, err := s.decodeStateParam(stateParam)
 	if err != nil {
 		return nil, fmt.Errorf("bad state param")
```

Comment:
How do we communicate that this is a User Error, and not a system error?

Reply:
same as https://github.com/gitpod-io/gitpod/pull/16331#discussion_r1102397434

That needs improvements.